openqabot/pc_helper.py

```diff
@@ -51,42 +24,36 @@ def get_latest_tools_image(query):
     return None
 
 
-# Applies PUBLIC_CLOUD_IMAGE_LOCATION based on the given PUBLIC_CLOUD_IMAGE_REGEX
-def apply_publiccloud_regex(settings):
-    try:
-        settings["PUBLIC_CLOUD_IMAGE_LOCATION"] = get_latest_pc_image(
-            settings["PUBLIC_CLOUD_IMAGE_REGEX"]
-        )
-        return settings
-    except BaseException as e:
-        log.warning(f"PUBLIC_CLOUD_IMAGE_REGEX handling failed: {e}")
-        settings["PUBLIC_CLOUD_IMAGE_LOCATION"] = None
-        return settings
-
-
 def apply_pc_tools_image(settings):
+    """
+    Use PUBLIC_CLOUD_TOOLS_IMAGE_QUERY to get latest tools image and set it into
+    PUBLIC_CLOUD_TOOLS_IMAGE_BASE
+    """
     try:
         if "PUBLIC_CLOUD_TOOLS_IMAGE_QUERY" in settings:
             settings["PUBLIC_CLOUD_TOOLS_IMAGE_BASE"] = get_latest_tools_image(
                 settings["PUBLIC_CLOUD_TOOLS_IMAGE_QUERY"]
             )
             del settings["PUBLIC_CLOUD_TOOLS_IMAGE_QUERY"]
-        return settings
     except BaseException as e:
         log.warning(f"PUBLIC_CLOUD_TOOLS_IMAGE_BASE handling failed: {e}")
-        return settings
+    return settings
 
 
-# Perform a pint query. Sucessive queries are cached
 @lru_cache(maxsize=None)
 def pint_query(query):
+    """
+    Perform a pint query. Sucessive queries are cached
+    """
     return requests.get(query).json()
 
 
-# Applies PUBLIC_CLOUD_IMAGE_LOCATION based on the given PUBLIC_CLOUD_IMAGE_REGEX
 def apply_publiccloud_pint_image(settings):
+    """
+    Applies PUBLIC_CLOUD_IMAGE_LOCATION based on the given PUBLIC_CLOUD_IMAGE_REGEX
+    """
+    pint_base_url = "https://susepubliccloudinfo.suse.com/v1/amazon/images/"
```

Comment:
But what about the others?

Reply:
somehow I was under impression that we using `  PUBLIC_CLOUD_PINT_QUERY:`  **only** for Amazon . after double-checking I find out that we also using it for google . thanks for hint I will correct this

Reply:
So I returned `PUBLIC_CLOUD_PINT_QUERY` back . seems like we can not avoid use of something defining provider and in that case it does not matter if it will be just "google"/"amazon" or just full url